versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5775 @@
+  }
+
+  output {
+    Array[String] matches = read_lines(stdout())
+  }
+}
+
+workflow wf {
+  input {
+    File infile
+    String pattern
+  }
+
+  call hello {
+    input: infile, pattern
+  }
+
+  output {
+    Array[String] matches = hello.matches
+  }
+}
+```
+
+This WDL document describes a task, called 'hello', and a workflow, called `wf`.
+
+* A `task` encapsulates a Bash script and a UNIX environment and presents them as a reusable function.
+* A `workflow` encapsulates a (directed, acyclic) graph of task calls that transforms input data to the desired outputs.
+
+Both workflows and tasks can accept input parameters and produce outputs. For example, the `wf` workflow has two input parameters, `File infile` and `String pattern`, and one output parameter, `Array[String] matches`. This simple workflow just passes its inputs to the `hello` task and returns its output.
+
+To execute this workflow, a WDL execution engine must be used (sometimes called the "WDL runtime" or "WDL implementation"). Some popular WDL execution engines are listed in the [README](https://github.com/openwdl/wdl#execution-engines). Along with the WDL file, the user must provide the implementation with values for the two input parameters. While implementations may provide their own mechanisms for launching workflows, all implementations minimally accept [inputs as JSON format](#json-input-format), which requires that inputs be fully qualified according to the namespacing rules described in the [Fully Qualified Names & Namespaced Identifiers](#fully-qualified-names--namespaced-identifiers) section. For example:
+
+|Variable     |Value    |
+|-------------|---------|
+|wf.pattern   |^[a-z]+$ |
+|wf.infile    |/file.txt|
+
+Or, in JSON format:
+
+```json
+{
+  "wf.pattern": "^[a-z]+$",
+  "wf.infile": "/file.txt"
+}
+```
+
+Running the `wf` workflow with these inputs would yield the following command line from the call to the `hello` task:
+
+```sh
+egrep '^[a-z]+$' '/file.txt'
+```
+
+Now imagine you want to run the `hello` task in parallel across many different input files. You can write a workflow to do so using the well-known [scatter-gather](https://en.wikipedia.org/wiki/Vectored_I/O#:~:text=In%20computing%2C%20vectored%20I%2FO,in%20a%20vector%20of%20buffers) pattern:
+
+```wdl
+workflow wf_parallel {
+  input {
+    Array[File] files
+    String pattern
+  }
+  
+  scatter (path in files) {
+    call hello {
+      input: 
+        infile = path,
+        pattern = pattern
+    }
+  }
+
+  output {
+    # WDL implicitly implements the 'gather' step, so
+    # the output of a scatter is always an array with
+    # elements in the same order as the input array
+    Array[Array[String]] all_matches = hello.matches
+  }
+}
+```
+
+The inputs to this workflow would be `wf_parallel.files` and `wf_parallel.pattern`. For example:
+
+```json
+{
+  "wf_parallel.pattern": "^[a-z]+$",
+  "wf_parallel.infile": ["/file1.txt", "/file2.txt"]
+}
+```
+
```

Comment:
Before moving on, it seems like I'm not sure what the goal of this first section was. It's too specific to just be a general orientation to the reader to what WDL is supposed to be.  Perhaps include here a link to other resources for finding example workflows, that is out of scope for this document, then move on to defining the spec?  Such as: https://github.com/openwdl/learn-wdl

Reply:
Thanks, I didn't know about that repo! Please read the revised version and see if it makes more sense.